multiqc/modules/slamdunk/slamdunk.py

```diff
@@ -0,0 +1,832 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Slamdunk """
+
+from __future__ import print_function
+import logging
+import re
+import numpy as np
+from distutils.version import StrictVersion
+from collections import OrderedDict
+
+from multiqc import config, BaseMultiqcModule, plots
+from curses.ascii import islower
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    Slamdunk module class, parses slamdunk logs.
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Slamdunk', anchor='slamdunk',
+        href='https://github.com/t-neumann/slamdunk',
+        info="is a tool to analyze SLAMSeq data. "\
+         "More info to come.")
+
+        self.slamdunk_data = dict()
+        
+        self.PCA_data = dict()
+        
+        self.utrates_data = dict()
+        
+        self.rates_data_plus = dict()
+        self.rates_data_minus = dict()
+        
+        self.nontc_per_readpos_plus = dict()
+        self.nontc_per_readpos_minus = dict()
+        
+        self.tc_per_readpos_plus = dict()
+        self.tc_per_readpos_minus = dict()
+        
+        self.nontc_per_utrpos_plus = dict()
+        self.nontc_per_utrpos_minus = dict()
+        
+        self.tc_per_utrpos_plus = dict()
+        self.tc_per_utrpos_minus = dict()
+
+        # Switch string for plots to produce
+        produce = "" 
+
+        # Check whether summary also contains number of counted reads
+        extendedSummary = True
+            
+        for f in self.find_log_files(config.sp['slamdunk']['summary'], filehandles = True):
+            extendedSummary = extendedSummary and self.parseSummary(f)
```

Comment:
Can it not leave the table cell blank instead of adding 0? In fact, I don't think you need this logic at all. If you give MultiQC a config section with a key that is not found in the data, it should just ignore it. So you can define `headers['counted']` and if none of the data has a key called `counted` then it will just be ignored. Otherwise, any samples that do have `counted` will have the data shown, any missing will just have an empty table cell.